### PR TITLE
[RFC][UN-13470] to-money: fixed display of left (and all) currencies

### PR DESCRIPTION
**Pull request by telmaantunes**
## Issue:
https://uniplaces.atlassian.net/browse/UN-13470

## Solution:
```js
export default function toMoney(amount, currencyCode, locale = 'pt-pt') {
  return Intl.NumberFormat(
    locale,
    {
      style: 'currency',
      currency: currencyCode,
      maximumFractionDigits: 0,
      minimumFractionDigits: 0,
      useGrouping: false
    }
  )
  .format(amount / 100)
  .replace(/\s/g, '');
}

toMoney(10000, 'EUR', 'pt-pt');
> '1000€'

toMoney(10000, 'EUR', 'en-gb');
> '€1000'
```

#### Euro convertions
screen shot 2018-02-06 at 12 14 18

#### Great Britain Pound conversions
screen shot 2018-02-06 at 12 15 07

#### United States Dollar conversions
screen shot 2018-02-06 at 12 16 02

#### Brazilian Real conversions
screen shot 2018-02-06 at 12 17 41

#### Search page examples
screen shot 2018-02-20 at 12 58 58
screen shot 2018-02-20 at 12 59 16

### CODE REVIEW

File: addon/utils/to-money.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import CurrencyTypeSymbol from 'ember-cli-uniq/enums/currency-symbol-type';
 
+const LEFT_DISPLAYED_CURRENCIES = ['GBP'];
+
 export default function toMoney(amount, currencyCode) {
-  return `${amount / 100}${CurrencyTypeSymbol[currencyCode]}`;
+  return LEFT_DISPLAYED_CURRENCIES.includes(currencyCode)
+    ? `${CurrencyTypeSymbol[currencyCode]}${amount / 100}`
```
Comment (DaniFrancisco) [inline addon/utils/to-money.js]: Just a suggestion, I think the result of the operation should be stored on a variable. Something like:

```javascript
let readableAmount = $amount / 100;

return LEFT_DISPLAYED_CURRENCIES.includes(currencyCode)
  ? `${CurrencyTypeSymbol[currencyCode]}${readableAmount}`
  : `${readableAmount}${CurrencyTypeSymbol[currencyCode]}`;
```

**Review by DaniFrancisco — DISMISSED** [review]

**Review by mmelo — DISMISSED** [review]
very much needed work, thanks 👌

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]
I think that, as discussed on Slack, this could be addressed in a more complete way from the beginning but it needs design approval :/

**Comment by AndreJoaquim** [comment]
EDIT: Transitioned this comment to the description of the PR

**Review by AndreJoaquim — DISMISSED** [review]

**Review by mmelo — DISMISSED** [review]

File: addon/helpers/to-money.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
-import toMoneyUtil from 'ember-cli-uniq/utils/to-money';
+import { isEmpty } from '@ember/utils';
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import toMoney from 'ember-cli-uniq/utils/to-money';
 
-export function toMoney([params]) {
-  return isEmpty(params) ? '' : toMoneyUtil(params.amount, params.currency_code);
-}
+const DEFAULT_LOCALE = 'pt-pt';
```
Comment (DaniFrancisco) [inline addon/helpers/to-money.js]: shouldn't the default be en-gb?

File: addon/utils/to-money.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
-import CurrencyTypeSymbol from 'ember-cli-uniq/enums/currency-symbol-type';
-
-export default function toMoney(amount, currencyCode) {
-  return `${amount / 100}${CurrencyTypeSymbol[currencyCode]}`;
+/* eslint new-cap: ["off"] */
+export default function toMoney(amount, currencyCode = 'EUR', locale = 'pt-pt') {
```
Comment (DaniFrancisco) [inline addon/utils/to-money.js]: same

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]
just asked some stuff

File: addon/helpers/to-money.js (hunk 2 above)
Comment (AndreJoaquim) [reply addon/helpers/to-money.js]: It's debatable... Where most of our users come from? :trollface:

Comment (AndreJoaquim) [reply addon/helpers/to-money.js]: But you're right! :)

Comment (DaniFrancisco) [reply addon/helpers/to-money.js]: I know you did it in `pt-pt` to maintain what we have today, but since our default is always `en-gb`...

**Review by DaniFrancisco — DISMISSED** [review]

**Review by mmelo — APPROVED** [review]

**Pull request merged**